ibft/storage/store.go

```diff
@@ -38,46 +41,151 @@ func init() {
 	}
 }
 
-// ibftStorage struct
+// participantStorage struct
 // instanceType is what separates different iBFT eth2 duty types (attestation, proposal and aggregation)
-type ibftStorage struct {
+type participantStorage struct {
 	prefix         []byte
+	oldPrefix      string // kept back for cleanup
 	db             basedb.Database
 	participantsMu sync.Mutex
 }
 
-// New create new ibft storage
-func New(db basedb.Database, prefix string) qbftstorage.QBFTStore {
-	return &ibftStorage{
-		prefix: []byte(prefix),
-		db:     db,
+// New create new participant store
+func New(db basedb.Database, prefix spectypes.BeaconRole) qbftstorage.ParticipantStore {
+	role := byte(prefix & 0xff)
+	return &participantStorage{
+		prefix:    []byte{role},
+		oldPrefix: prefix.String(),
+		db:        db,
 	}
 }
 
-// CleanAllInstances removes all StoredInstance's & highest StoredInstance's for msgID.
-func (i *ibftStorage) CleanAllInstances(msgID []byte) error {
-	prefix := i.prefix
-	prefix = append(prefix, msgID[:]...)
-	prefix = append(prefix, []byte(instanceKey)...)
-	err := i.db.DropPrefix(prefix)
+// InitialSlotGC waits for the initial tick and then removes all slots below the tickSlot - retain
+func (i *participantStorage) InitialSlotGC(ctx context.Context, logger *zap.Logger, threashold phase0.Slot) {
+	logger.Info("start initial stale slot cleanup", zap.String("store", i.ID()), fields.Slot(threashold))
+
+	// remove ALL slots below the threashold
+	start := time.Now()
+	count := i.removeSlotsOlderThan(logger, threashold)
+
+	logger.Info("removed stale slot entries", zap.String("store", i.ID()), fields.Slot(threashold), zap.Int("count", count), zap.Duration("took", time.Since(start)))
+}
+
+// SlotGC on every tick looks up and removes the slots that fall below the retain threashold
+func (i *participantStorage) SlotGC(ctx context.Context, logger *zap.Logger, slotTickerProvider slotticker.Provider, retain int) {
+	ticker := slotTickerProvider()
+	logger.Info("start stale slot cleanup loop", zap.String("store", i.ID()))
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.Next():
+			threashold := ticker.Slot() - phase0.Slot(retain) - 1 // #nosec G115
+			count, err := i.removeSlotAt(threashold)
+			if err != nil {
+				logger.Error("remove slot at", zap.String("store", i.ID()), fields.Slot(threashold))
+			}
```

Comment:
I don't think that makes any sense, don't we want to keep all history in exporter? even if we want to gc or remove old slots it must be really far in the past. it probably makes more sense to come up with some kinds of compaction format than removing the data. 

@moshe-blox @MatusKysel @liorrutenberg 
do you guys think it makes sense to say exporter supports storing data up until X slots in the past?